src/main/kotlin/io/github/enyason/client/ReplicateClient.kt

```diff
@@ -0,0 +1,38 @@
+package io.github.enyason.client
+
+import io.github.enyason.io.github.enyason.predictable.validate
+import io.github.enyason.predictable.Predictable
+import io.github.enyason.predictions.PredictionsApi
+import io.github.enyason.predictions.createPrediction
+
+class ReplicateClient(
+    private val predictionAPI: PredictionsApi
+) : Replicate {
+
+    override suspend fun createPrediction(predictable: Predictable): Result<Any?> {
+        predictable.validate()
```

Comment:
Can we move this into the try block so that the exception (if any) can also be wrapped in Result.failure?

Reply:
okay, thats okay